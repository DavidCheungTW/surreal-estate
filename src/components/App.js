import React, { useEffect, useState } from "react";
import { Routes, Route, useLocation } from "react-router-dom";
import NavBar from "./NavBar";
import Properties from "./Properties";
import AddProperty from "./AddProperty";
import "../styles/app.css";

const App = () => {
  const location = useLocation();
  const [menu, setMenu] = useState("classVP");

  useEffect(() => {
    switch (location.pathname) {
      case "/":
        setMenu("classVP");
        break;
      case "/add-property":
        setMenu("classAP");
        break;
      default:
        setMenu("");
    }
  }, [location]);

  return (
    <div className="App">
      <h2>Surreal Estate</h2>
      <NavBar menu={menu} />
      <Routes>
        <Route path="/" element={<Properties />} />
        <Route path="add-property" element={<AddProperty />} />
      </Routes>
    </div>
  );
};

export default App;
