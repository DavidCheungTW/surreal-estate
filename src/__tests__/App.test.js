import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import App from "../components/App";

describe("App", () => {
  test("Renders as expected", () => {
    const rendered = renderer.create(
      <Router>
        <App />
      </Router>
    );

    expect(rendered).toMatchSnapshot();
  });

  test("Assert Surreal Estate Heading", () => {
    render(
      <Router>
        <App />
      </Router>
    );

    expect(screen.getByText(/Surreal Estate/i)).toBeTruthy();
    expect(screen.getByText(/Surreal Estate/i)).toBeInstanceOf(
      HTMLHeadingElement
    );
  });
});
