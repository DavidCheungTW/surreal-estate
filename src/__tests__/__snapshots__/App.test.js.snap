// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Renders as expected 1`] = `
<div
  className="App"
>
  <h2>
    Surreal Estate
  </h2>
  <div
    className="navbar"
  >
    <img
      alt="logo"
      className="navbar-logo"
      src="logo.png"
    />
    <ul
      className="navbar-links"
    >
      <li
        className="navbar-links-item"
      >
        <a
          className="item"
          href="/"
          onClick={[Function]}
        >
          View Properties
        </a>
      </li>
      <li
        className="navbar-links-item"
      >
        <a
          className="item"
          href="/add-property"
          onClick={[Function]}
        >
          Add a Property
        </a>
      </li>
    </ul>
  </div>
  <div
    className="properties"
  >
    <h3>
      Properties Page
    </h3>
  </div>
</div>
`;
